src/common/dump.cpp

```diff
@@ -230,3 +230,54 @@ int dump_hardfault_test_1(void) {
     volatile int c = 1 / b;
     return c;
 }
+
+// Dumping error message
+
+void dump_err_to_xflash(const char *error, const char *title) {
+    // This is called after w25x initialization (in dumpinfo_t dumping), so no additional w25x_init() is needed
+    uint8_t valid_flag = dump_err_in_xflash_get_valid_flag();
+    if (valid_flag != 0xFF) {                                       // ==> message space is not empty
+        if (valid_flag != 0 || dump_err_in_xflash_is_displayed()) { // ==> erase sector if flag is corrupted (!= 0x00) or is valid and is displayed
+            w25x_sector_erase(w25x_error_start_adress);
+            w25x_fetch_error();
+        }
+    }
```

Comment:
I would suggest to remove all those conditions and simply always call w25x_sector_erase(w25x_error_start_adress).

Alternatively if you really want to avoid that one in printer lifetime redundant w25x_sector_erase call which happens only when newly manufactured printer wants to show error message for the first time. You would need to do this properly - read whole to be written area and check if the whole area is erased (contains 0xff).

Reply:
Yep, that will be a lot easier. It means, that printer will show the latest redscreen, not the first occured, but I'm fine with that. `9d28bd0`